### PR TITLE
Allow nb-controller get/list/watch/create events

### PR DESCRIPTION
**Pull request by fediazgon**
**Which issue is resolved by this Pull Request:**

Partial fix of https://github.com/kubeflow/kubeflow/issues/3772, used by already merged PR https://github.com/kubeflow/kubeflow/pull/4139

**Description of your changes:**

The `notebook-controller` reissues pods and statefulsets events as notebook events, hence, it needs permission to get/list/watch/create events otherwise:

```
Failed to list *v1.Event: events is forbidden: User "system:serviceaccount:kubeflow:notebook-controller-service-account" cannot list resource "events" in API group "" at the cluster scope
```

**Checklist:**
- [x] Unit tests have been rebuilt: 
    1. `cd manifests/tests`
    2. `make generate`
    3. `make test`

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/manifests/641)

### CODE REVIEW

**Comment by fediazgon** [comment]
@krishnadurai The reissued events are listed by `jupyter-web-app` to give the user extra information about why the notebook is not starting. I added those permissions in https://github.com/kubeflow/manifests/pull/641/commits/d7dfe0dbf0be62b9de8b10b27c14fc2edb6f756b

File: jupyter/jupyter-web-app/base/cluster-role.yaml [hunk 1]
```diff
@@ -31,6 +31,21 @@ rules:
...
+  - get
+  - list
+  - watch
+  - create
+- apiGroups:
+  - ""
+  resources:
+  - events
+  verbs:
+  - list
```
Comment (krishnadurai) [inline jupyter/jupyter-web-app/base/cluster-role.yaml]: Isn't this just a repetition of the above?

Comment (fediazgon) [reply jupyter/jupyter-web-app/base/cluster-role.yaml]: Sorry, didn't get it right the first time either. `jupter-web-app` should be able to "list" and `notebook-controller` should be able to "get/list/watch/delete". I mistakenly edited "jupyter/jupyter-web-app" but left "jupyter/notebook-controller". I amended both commits.

Comment (krishnadurai) [reply jupyter/jupyter-web-app/base/cluster-role.yaml]: Ok, thanks.

**Comment by jlewi** [comment]
Drive by approve.

/approve

Filed kubeflow/manifests#677 to create an appropriate OWNERs file.

**Pull request merged**